dev-docs/design/ssr-react.md

```diff
@@ -0,0 +1,96 @@
+
+The default behavior of the element renderer here will be to set all props provided as attributes on the element as they're server rendered to match the base React client behavior. However, if the prop named `_$litProps$` is passed, this object will contain properties that will be set as properties on the element by the renderer. This will also cause the host element to have the `defer-hydration` attribute so that hydration can be paused on the client until all properties are set on it.
+
+#### Patching `createElement`
+
+The classic JSX transform compiles JSX to `React.createElement` calls. Our `@lit-labs/react` wrapper also uses the `React` passed in by user to call `createElement` on it.
+
+We can provided an alternate `createElement` function that applies the core strategy above.
+
+For convenience, we can provide a side-effectful import that monkey patches `React.createElement` to be this alternate version.
+
+Using export conditions, we can create a module `@lit-labs/ssr-react/enable-lit-ssr.js` that when imported in Node will monkey patch `React.createElement` and when imported in the browser will load the `@lit-labs/ssr-client/lit-element-hydrate-support.js`.
+
+#### Using runtime JSX
+
+The runtime JSX transform compiles JSX to various jsx function calls, importing `jsx` or `jsxs` from `<package>/jsx-runtime` or `jsxDEV` from `<package>/jsx-dev-runtime` depending on the environment and whether the `children` array was created statically.
+
+We can provide alternate jsx functions that enhance the base React jsx functions with the core strategy above.
+
+This is mainly for targeting bare custom elements as wrapped components must have `React.createElement` patched to work.
+
+### `@lit-labs/react` package
+
+The wrapper made with `createComponent` from the Node build of this package will check whether `React.createElement` has been patched by checking its `.name` patches ours. If so, it will pass element properties in a special prop named `_$litProps$` to `createElement` to be used for server rendering as described above.
+
+On the client, the wrapper on mount will set properties on the element and remove the `defer-hydration` attribute to start Lit element hydration. It will also add `suppressHydrationWarning` to the prop of the host element to suppress React's warning for having extra attributes on the server rendered HTML like `defer-hydration`.
+
+There was consideration of baking in the whole SSR template shadowroot generation into the Node build of `@lit-labs/react` package itself which would remove the need for monkey patching `createElement`, but users would then have to make sure to manually load the `lit-element-hydrate-support` early anyway. It's not out of the realms of possiblity.
```

Comment:
"It's not out of the realms of possiblity." - does this mean this could still be an additional enhancement after this change, and is compatible with this change?